### ISSUE TITLE
Callback memory leak

### ISSUE THREAD

**Issue by Charl-AI** [Charl-AI/lightning-template#7]
**Severity: [high]**

**Describe the bug**
Memory leak in model where all step outputs are stored. This causes every image,target and logits, tensor from an epoch to be cached, so memory increases linearly over an epoch. Crashes when working with larger images.

**Expected behavior**
Only tensors from a single batch should be cached. Memory used should remain roughly constant throughout an epoch.

**Screenshots**
![image](https://user-images.githubusercontent.com/46627645/134179498-ec96a434-3c4c-410d-8689-8c0d4bcd7e74.png)
